packages/ui/src/ui/pages/query-tracker/module/query/reducer.ts

```diff
@@ -89,12 +92,15 @@ export function reducer(state = initState, action: Actions): QueryState {
             };
         }
         case SET_QUERY_PATCH: {
```

Comment:
It seems to me that those two actions SET_QUERY_PATCH/SET_QUERY cover all of our cases.

Reply:
All changed parts of the state that received from api should prevent unload. I nearly sure that some data for SET_QUERY_PARAMS comes from api. Could you please check what parts of the state comes from api?

Reply:
It seems that SET_QUERY_PARAMS only works with the GET parameters of the page.